cpp/src/io/orc/reader_impl.cu

```diff
@@ -464,69 +466,55 @@ class aggregate_orc_metadata {
 
   /**
    * @brief Adds column as per the request and saves metadata about children.
-   *        Struct children are in the same level as struct, only list column
-   *        children are pushed to next level.
+   *        Children of a column will be added to the next level.
    *
    * @param selection A vector that saves list of columns as per levels of nesting.
    * @param types A vector of schema types of columns.
    * @param level current level of nesting.
    * @param id current column id that needs to be added.
    * @param has_timestamp_column True if timestamp column present and false otherwise.
-   *
-   * @return returns number of child columns at same level in case of struct and next level in case
-   * of list
+   * @param has_nested_column True if any of the selected column is a nested type.
    */
-  uint32_t add_column(std::vector<std::vector<orc_column_meta>>& selection,
-                      std::vector<SchemaType> const& types,
-                      const size_t level,
-                      const uint32_t id,
-                      bool& has_timestamp_column,
-                      bool& has_list_column)
+  void add_column(std::vector<std::vector<orc_column_meta>>& selection,
+                  std::vector<SchemaType> const& types,
+                  const size_t level,
+                  const uint32_t id,
+                  bool& has_timestamp_column,
+                  bool& has_nested_column)
   {
-    uint32_t num_lvl_child_columns = 0;
     if (level == selection.size()) { selection.emplace_back(); }
     selection[level].push_back({id, 0});
     const int col_id = selection[level].size() - 1;
     if (types[id].kind == orc::TIMESTAMP) { has_timestamp_column = true; }
 
     switch (types[id].kind) {
```

Comment:
don't really need the switch here, there's only one case.

Reply:
I was thinking I might require while adding map support, but I will change it while adding it.

Reply:
feel free to keep, then. Not important change at all.